tikv-client/src/main/java/com/pingcap/tikv/TiSession.java

```diff
@@ -187,12 +189,17 @@ public void injectCallBackFunc(Function<CacheInvalidateEvent, Void> callBackFunc
   }
 
   public static void clearCache() {
-    TiSession.sessionCachedMap.clear();
+    synchronized (sessionCachedMap) {
+      TiSession.sessionCachedMap.clear();
+    }
   }
 
   @Override
-  public synchronized void close() throws Exception {
-    sessionCachedMap.remove(conf.getPdAddrsString());
+  public void close() throws Exception {
```

Comment:
maybe we should remain close() synchronized?

Reply:
fixed